### PR TITLE
Add new and create for admin words

### PR DESCRIPTION
**Pull request by kian1213**
Hello, please review my pull request. Thanks

### CODE REVIEW

**Comment by codegracie** [comment]
Okay.

File: app/controllers/admin/words_controller.rb [hunk 1]
```diff
@@ -6,4 +6,24 @@ def index
...
+    if @word.save
+      redirect_to admin_words_path
+    else
+      render "new"
+    end
+  end
+
+  private
+  def word_params
+    params.require(:word).permit(:content, :category_id)
```
Comment (dieunb) [inline app/controllers/admin/words_controller.rb]: remove `()`

File: app/models/word.rb [hunk 2]
```diff
@@ -3,5 +3,5 @@ class Word < ActiveRecord::Base
   has_many :answers
   belongs_to :category
 
-  validates :content, presence: true
+  validates :content, :category_id, presence: true
```
Comment (dieunb) [inline app/models/word.rb]: difference validate between `category` and `category_id`? And what do you choice?

File: app/models/word.rb [hunk 3]
```diff
@@ -1,7 +1,9 @@
 class Word < ActiveRecord::Base
   has_many :lesson_words
-  has_many :answers
+  has_many :answers, dependent: :destroy
   belongs_to :category
+  accepts_nested_attributes_for :answers
```
Comment (MuhammadTamzid) [inline app/models/word.rb]: use new line before accepts_nested_attributes_for :answers

File: app/views/admin/words/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  </div>
+  <div class="form-group">
+    <%= f.label :category_id, "Category", class: "col-sm-2 control-label" %>
+    <div class="col-sm-10">
+      <%= f.collection_select :category_id, @categories, :id, :name,
+        {prompt: "Select a category"}, {class: "form-control"} %>
+    </div>
+  </div>
+  <%= t ".list_of_answers" %>
+  <% 4.times do %>
```
Comment (MuhammadTamzid) [inline app/views/admin/words/_form.html.erb]: I think this is not good.
If admin want to add more than 4 answers, then??

File: app/views/admin/words/_answer_fields.html.erb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  <%= f.label :content, "Answer", class: "col-sm-2 control-label" %>
+  <div class="col-sm-4">
+    <%= f.text_field :content, class: "form-control" %>
+  </div>
+  <%= f.label :correct, class: "col-sm-2 control-label" %>
+  <div class="col-sm-2">
+    <%= f.check_box :correct, class: "form-control" %>
+  </div>
+  <div class="col-sm-2">
+    <a href="#", class="btn btn-danger", id="remove-answer">
```
Comment (MuhammadTamzid) [inline app/views/admin/words/_answer_fields.html.erb]: you can use "link_to_remove_fields" here

File: app/models/word.rb [hunk 6]
```diff
@@ -1,7 +1,10 @@
 class Word < ActiveRecord::Base
   has_many :lesson_words
-  has_many :answers
+  has_many :answers, dependent: :destroy
   belongs_to :category
 
+  accepts_nested_attributes_for :answers, reject_if: lambda {|a| a[:content].blank?}, allow_destroy: true
```
Comment (MuhammadTamzid) [inline app/models/word.rb]: 80 char

File: app/controllers/admin/words_controller.rb [hunk 7]
```diff
@@ -6,4 +6,24 @@ def index
...
+    if @word.save
+      redirect_to admin_words_path
+    else
+      render "new"
+    end
+  end
+
+  private
+  def word_params
+    params.require(:word).permit :content, :category_id, answers_attributes: [:id, :word_id, :content, :correct]
```
Comment (MuhammadTamzid) [inline app/controllers/admin/words_controller.rb]: 80 char

File: app/views/admin/words/_answer_fields.html.erb [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<div class="form-group fields">
+  <%= f.hidden_field :id, value: @answer_id %>
+  <%= f.label :content, "Answer", class: "col-sm-2 control-label" %>
```
Comment (dieunb) [inline app/views/admin/words/_answer_fields.html.erb]: i18n

File: app/assets/javascripts/admin/styles.js [hunk 9]
```diff
@@ -1,4 +1,15 @@
...
+  $("#wrapper").toggleClass("toggled");
+});
+
+function remove_fields(link) {
+  $(link).parent().children('.removable')[0].value = 1;
+  $(link).parent().parent().hide();
+}
+
+function add_fields(link, association, content) {
+    var new_id = new Date().getTime();
```
Comment (dieunb) [inline app/assets/javascripts/admin/styles.js]: if you use indent = 2 spaces, pls use it for this case

**Pull request merged**